src/postgres/interface.c

```diff
@@ -165,6 +165,23 @@ pgDbIsSystemId(const unsigned int id)
     FUNCTION_TEST_RETURN(BOOL, id < FirstNormalObjectId);
 }
 
+/**********************************************************************************************************************************/
+FN_EXTERN bool
+pgWalSegmentSizeValid(const unsigned int pgVersion, const size_t walSegmentSize)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM(UINT, pgVersion);
+        FUNCTION_TEST_PARAM(SIZE, walSegmentSize);
+    FUNCTION_TEST_END();
+
+    if (pgVersion < PG_VERSION_10 && walSegmentSize > PG_9_MAX_WAL_SEGMENT_SIZE)
+    {
+        FUNCTION_TEST_RETURN(BOOL, false);
+    }
+
+    FUNCTION_TEST_RETURN(BOOL, IsValidWalSegSize(walSegmentSize));
+}
+
```

Comment:
I don't think it is worth adding this exception for EOL versions of PostgreSQL. If somebody somehow convinces older versions of PostgreSQL to use large WAL segments sizes, then that is on them.